src/H5Fint.c

```diff
@@ -1752,16 +1817,102 @@ H5F_open(const char *name, unsigned flags, hid_t fcpl_id, hid_t fapl_id)
     unsigned           page_buf_min_meta_perc = 0;
     unsigned           page_buf_min_raw_perc  = 0;
-    hbool_t            set_flag               = FALSE; /*set the status_flags in the superblock */
-    hbool_t            clear                  = FALSE; /*clear the status_flags         */
-    hbool_t            evict_on_close;                 /* evict on close value from plist  */
-    hbool_t            use_file_locking = TRUE;        /* Using file locks? */
-    hbool_t            ci_load          = FALSE;       /* whether MDC ci load requested */
-    hbool_t            ci_write         = FALSE;       /* whether MDC CI write requested */
-    H5F_t *            ret_value        = NULL;        /*actual return value           */
+    hbool_t            vfd_swmr               = FALSE;    /* TRUE iff opening file with VFD SWMR */
+    hbool_t            vfd_swmr_writer        = FALSE;    /* TRUE iff opening file as VFD SWMR   */
+                                                          /* writer.                             */
+    hbool_t pop_vfd_swmr_reader_vfd = FALSE;              /* Flag set when the VFD SWMR reader VFD */
+                                                          /* has been pushed on the supplied fapl  */
+                                                          /* and must be popped before return.      */
+    hbool_t                 set_flag = FALSE;             /* Set the status_flags in the superblock */
+    hbool_t                 clear    = FALSE;             /* Clear the status_flags */
+    hbool_t                 evict_on_close;               /* Evict on close value from plist */
+    hbool_t                 use_file_locking    = TRUE;   /* Using file locks? */
+    hbool_t                 ci_load             = FALSE;  /* Whether MDC ci load requested */
+    hbool_t                 ci_write            = FALSE;  /* Whether MDC ci write requested */
+    hbool_t                 file_create         = FALSE;  /* Creating a new file or not */
+    H5F_vfd_swmr_config_t * vfd_swmr_config_ptr = NULL;   /* Points to VFD SMWR config info */
+    H5F_generate_md_ck_cb_t cb_info             = {NULL}; /* For VFD SWMR NFS testing:
+                                                             initialize the callback to generate
+                                                             checksums for metadata files */
+    H5F_t *ret_value = NULL;                              /* Actual return value */
 
     FUNC_ENTER_NOAPI(NULL)
 
+    /* Get the file access property list, for future queries */
+    if (NULL == (a_plist = (H5P_genplist_t *)H5I_object(fapl_id)))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, NULL, "not file access property list")
+
+    /* start by testing to see if we are opening the file VFD SWMR reader.  If
+     * we are, we must "push" the vfd swrm reader vfd on the vfd "stack" supplied
+     * by the user in the fapl.  Since the user may use the fapl elsewhere, we
```

Comment:
This is not how the VFD stack operates and hard-wires the SWMR VFD into this part of the library, which it should be independent of.   The SWMR VFD should be changed to be a pass-through VFD (modeled on the other pass-through VFDs).